src/components/RequestBox/RequestBox.js

```diff
@@ -40,6 +40,24 @@ export default class RequestBox extends Component {
     this.renderError = this.renderError.bind(this);
   }
 
+  replaceRequestAndSubmit(request) {
+    console.log("replaceRequestAndSubmit");
+
+    // replace the request in the state with the new one
+    if (request.resourceType.toUpperCase() === "MEDICATIONREQUEST") {
+      this.state.medicationRequest = request;
+    } else if (request.resourceType.toUpperCase() === "DEVICEREQUEST") {
+      this.state.deviceRequest = request;
+    } else if (request.resourceType.toUpperCase() === "SERVICEREQUEST") {
+      this.state.serviceRequest = request;
+    } else if (request.resourceType.toUpperCase() === "MEDICATIONDISPENSE") {
+      this.state.medicationDispense = request;
+    }
+
+    // submit the CRD request
+    this.submit();
```

Comment:
changing the above to use `setState` will break this, since the submit function uses the state and the state won't be updated by the time this function is called.  I'd suggest breaking up the submit function into two parts, one for generating the prefetch, and the other for actually submitting, and then use the prop function `submit_info` in this function to actually do the submitting.

So, it would look something like
```
const prefetch = this.makePrefetch(request) // new function
this.props.submitInfo(prefetch, request, this.state.patient)
```
The original `submit` function should just be modified to use the new prefetch function, but otherwise remain the same.

Reply:
done, check it over